diff --git a/node_modules/astro/dist/vite-plugin-scripts/index.js b/node_modules/astro/dist/vite-plugin-scripts/index.js
index 37b467b..aca6380 100644
--- a/node_modules/astro/dist/vite-plugin-scripts/index.js
+++ b/node_modules/astro/dist/vite-plugin-scripts/index.js
@@ -38,7 +38,7 @@ function astroScriptsPlugin({ settings }) {
     },
     buildStart() {
       const hasHydrationScripts = settings.scripts.some((s) => s.stage === "before-hydration");
-      if (hasHydrationScripts && (this.environment.name === ASTRO_VITE_ENVIRONMENT_NAMES.prerender || this.environment.name === ASTRO_VITE_ENVIRONMENT_NAMES.ssr)) {
+      if (hasHydrationScripts && (this.environment.name === ASTRO_VITE_ENVIRONMENT_NAMES.client || this.environment.name === ASTRO_VITE_ENVIRONMENT_NAMES.prerender || this.environment.name === ASTRO_VITE_ENVIRONMENT_NAMES.ssr)) {
         this.emitFile({
           type: "chunk",
           id: BEFORE_HYDRATION_SCRIPT_ID,
